### PR TITLE
Change evaluation times to include the final time.

### PR DESCRIPTION
**Pull request by sebgrijalva**
This addresses #249.

### CODE REVIEW

**Comment by sebgrijalva** [comment]
For the moment I changed the way `Simulation.evaluation_times` is assigned. Notice also that `Simulation._times` is changed to `Simulation._sampling_indexes` (although it keeps the factor of 1/1000 so far), since these indexes are used to write the samples from `Sequence`

**Comment by HGSilveri** [comment]
> For the moment I changed the way `Simulation.evaluation_times` is assigned. Notice also that `Simulation._times` is changed to `Simulation._sampling_indexes` (although it keeps the factor of 1/1000 so far), since these indexes are used to write the samples from `Sequence`

Wouldn't `sampling_times` be more appropriate than `sampling_indexes`?

**Comment by sebgrijalva** [comment]
> > For the moment I changed the way `Simulation.evaluation_times` is assigned. Notice also that `Simulation._times` is changed to `Simulation._sampling_indexes` (although it keeps the factor of 1/1000 so far), since these indexes are used to write the samples from `Sequence`
> 
> Wouldn't `sampling_times` be more appropriate than `sampling_indexes`?

So far what's happening is that `Simulation._sampling_times` stores the indexes from the Sequence that are consistent with the sampling rate and that will be used to construct the `Qutip.QobjEvo` hamiltonian... and then divides by 1000, so it's Ok to call them "times" in microseconds.

Then, the solver itself is using a `tlist` consisting on an array from 0 to `_tot_duration` (now including it) mutliplied by the sampling rate. This seems more adapted to be called a "time", although it loses connection with the Sequence samples.

Finally, the evaluation times are being fed to the `SimResults` object to give the output that the user wanted (e.g. if "Minimal", gives two states, two expectation values, etc.)

**Comment by sebgrijalva** [comment]
I see that some notebooks are being affected by these changes, most likely because they use `Simulation._times`at some point. I leave this note here for myself when I come back to this

File: pulser/simulation/simulation.py [hunk 1]
```diff
@@ -886,15 +887,25 @@ def _run_solver() -> CoherentResults:
                 result = qutip.mesolve(
                     liouvillian,
                     self.initial_state,
-                    self._eval_times_array,
+                    np.linspace(
+                        0,
+                        self._tot_duration,
+                        int(self._sampling_rate * self._tot_duration) + 1,
+                    )
+                    / 1000,
```
Comment (HGSilveri) [inline pulser/simulation/simulation.py:895]: I don't get this one... Why not keep using `self._eval_times_array`?

File: pulser/tests/test_simresults.py [hunk 2]
```diff
@@ -210,7 +219,9 @@ def test_expect():
     sim_single.set_config(config)
     sim_single.evaluation_times = "Minimal"
     results_single = sim_single.run()
+    print("states after run are:", results_single.states)
     exp = results_single.expect(op)[0]
+    print("expect results are", exp)
```
Comment (HGSilveri) [inline pulser/tests/test_simresults.py:224]: I guess you can remove these

File: pulser/tests/test_simresults.py [hunk 3]
```diff
@@ -284,22 +294,30 @@ def test_sample_final_state():
 
 
 def test_sample_final_state_noisy():
+    # Sample final state of noisy simulation and check resulting Counter:
+    print(results_noisy.states[-1])
+    print(results_noisy.get_final_state())
```
Comment (HGSilveri) [inline pulser/tests/test_simresults.py:299]: And these... The `print` statements in general

**Review by HGSilveri — COMMENTED** [review]
I guess I'm a bit puzzled about some of the changes...

**Comment by sebgrijalva** [comment]
Closing this for a fresh start on #282

**Pull request closed without merging**